Добрый день!

Lidia128 opened the pull request:
Написала программу, но файл в котором я могу вносить данне не такой как на видео. Что-то не так делаю, но не могу понять что. Помогите пожалуйста разобраться.

Lidia128 committed:
Добрый день!
Написала программу, но файл в котором я могу вносить данне не такой как на видео. Что-то не так делаю, но не могу понять что. Помогите пожалуйста разобраться.

DmitryN10 commented on src/main/java/me/recipe/homework/controller/IngredientController.java:
```diff
@@ -0,0 +1,25 @@
+
+    public IngredientController(IngredientService ingredientService) {
+        this.ingredientService = ingredientService;
+    }
```
```
  @GetMapping
    public Collection<Ingredient> getAllIngredient() {
        return ingredientService.getAll();
    }

```

DmitryN10 replied:
не хвататет такого метода

DmitryN10 commented on src/main/java/me/recipe/homework/controller/RecipeController.java:
```diff
@@ -0,0 +1,16 @@
+@RequestMapping("/recipe")
+
+public class RecipeController {
+    private final RecipeService recipeService;
```
```
@GetMapping
    public Collection<Recipe> getAllRecipe() {
        return recipeService.getAll();
    }
```

DmitryN10 commented on src/main/java/me/recipe/homework/service/IngredientService.java:
```diff
@@ -0,0 +1,34 @@
+
+@Service
+
+public class IngredientService {
```
стоило под каждый сервис также создать интерфейс

Lidia128 committed:
Скорреткировала программу. Прошу проверить.

Lidia128 committed:
Добрый день!
Прошу помочь скорректировать, не могу почему подчеркивает красным.В recipeController в строке 33 recipe и в RecipeServiceImpl строка 40 id.

OlgaLugacheva commented on src/main/java/me/recipe/homework/controller/RecipeController.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Recipe> editRecipe(@PathVariable long id, @RequestBody Recipe recipe) {
```
в карте у вас Integer - тогда и зедсь тип id должен быть Integer

OlgaLugacheva commented on src/main/java/me/recipe/homework/impl/RecipeServiceImpl.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    }
+    @Override
+    public Recipe editRecipe(long id, Recipe recipe) {
```
Здесь тоже используйте Integer

OlgaLugacheva commented on src/main/java/me/recipe/homework/controller/RecipeController.java:
```diff
@@ -0,0 +1,51 @@
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Recipe> editRecipe(@PathVariable long id, @RequestBody Recipe recipe) {
+        Recipe recipe = recipeService.editRecipe(id, recipe);
```
определите новую пеерменную 
 Recipe updatedRecipe = recipeService.editRecipe(id, recipe);

Lidia128 committed:
Добрый день!
Внесла правки. Прошу проверить.

OlgaLugacheva commented on src/main/java/me/recipe/homework/controller/RecipeController.java:
```diff
@@ -30,7 +30,7 @@ public Recipe addRecipe(@RequestBody Recipe recipe) {
-        Recipe recipe = recipeService.editRecipe(id, recipe);
+        Recipe updatedRecipe = recipeService.editRecipe(id, recipe);
         if (recipe == null) {
             return ResponseEntity.notFound().build();
```
в респонс тогда тоже нужно updatedReceipe

Lidia128 committed:
Доработала
